File: Design_Pattern_and_Principles/DecoratorPatternExample/DecoratorTest.java
public class DecoratorTest {
    public static void main(String[] args) {
        Notifier engine = new EmailNotifier();

        engine = new SMSNotifierDecorator(engine);

        engine = new SlackNotifierDecorator(engine);

        System.out.println("Dispatching combined update stream:");
        engine.send("Build #42 is deploying now.");
    }
}

File: Design_Pattern_and_Principles/DecoratorPatternExample/DecoratorTest.cs
using System;

public class DecoratorTest {
    public static void Main(string[] args) {
        INotifier engine = new EmailNotifier();
        engine = new SMSNotifierDecorator(engine);
        engine = new SlackNotifierDecorator(engine);

        Console.WriteLine("Dispatching combined update stream:");
        engine.Send("Build #42 is deploying now.");
    }
}